### DISCUSSION TITLE
Why does FFCV use extra memory on main GPU in distributed mode?

### DISCUSSION THREAD

**Discussion by numpee** [libffcv/ffcv#129] [Q&A]
When training on ImageNet with FFCV, I've noticed that additional processes will spawn on GPU:0 that use up some memory. The additional processes spawn at the first iteration, if I haven't previously cached the dataset onto RAM. 
Below is a screenshot of my `nvidia-smi` output when loading data via FFCV, loading a model to GPU, but not actually running training (no forward or backward pass through the model):
![image](https://user-images.githubusercontent.com/29429992/152504683-9c58a157-9abd-4e3f-a2dd-896517143806.png)
In the figure above, processes 49846, 49847, and 49848 on GPU:0 are spawned at the first iteration.

Based on the behavior, I assume the caching operation uses some GPU memory. Is this correct? Also, is there a way to disable this?
The issue is that I want to use a specific batch size, but will run into OOM because of the additional memory used. 

FFCV version: 0.0.2

**Comment by GuillaumeLeclerc** [comment]
This should be fixed in the next release. Have you tried v0.0.3 ?

**Comment by numpee** [comment]
Thanks for the reply, @GuillaumeLeclerc.
I just updated to 0.0.3rc1 (seems like this is the latest release on PyPI?) and the issue still persists.
I've tested on 2 separate machines. 
Perhaps PyPI hasn't been updated with 0.0.3?

Edit 1: I just noticed that this was mention in [this pull request](https://github.com/libffcv/ffcv/pull/115). Will try to install from the v0.0.4 branch

Edit 2: Ok I manually installed v0.0.4 from the git repo. Seems to be fixed now - sorry for the hassle 😢
